rpmlb/builder/koji.py

```diff
@@ -0,0 +1,173 @@
+"""Builder interface for the Koji build service."""
+
+import re
+from pathlib import Path
+from typing import Iterator, Optional, Sequence
+
+from ..utils import run_cmd, run_cmd_with_capture
+from ..work import Work
+from .base import BaseBuilder
+
+
+class KojiBuilder(BaseBuilder):
+    """Builder interface for the Koji build service."""
+
+    DEFAULT_TARGET_FORMAT = 'sclo{epel}-{collection}-rh-el{epel}'
+
+    def __init__(self):
+        """Initialize the slots"""
+
+        self.collection = None  #: The collection to build
+        self.epel = None  #: Version of EPEL to build for
+        self.profile = None  #: The koji configuration profile to use
+        #: The format string for construction of the actual template name
+        self.target_format = type(self).DEFAULT_TARGET_FORMAT
+
+    @property
+    def base_command(self) -> Sequence[str]:
```

Comment:
Keeping the "public" interface of all builder classes the same and making them all interchangable is good design. However an underscored method does not only indicate this is not part of the API, to me it also indicates "this is and implementation detail that may change at any time". Adding more "public" (= not underscored) methods is a promis for anyone who wants to subclass this that they can override those methods safely. I.e. I see two kinds of API here: one for a code that uses the builder class (and that one should remain interchengable) and one for any possible subclasses of this given builder. By adding extra methods to extend the second API you are not necessary breaking the first one.